numpy/core/src/multiarray/descriptor.c

```diff
@@ -3166,10 +3168,25 @@ arraydescr_struct_str(PyArray_Descr *dtype, int includealignflag)
      */
     if (!(includealignflag && (dtype->flags&NPY_ALIGNED_STRUCT)) &&
                         is_dtype_struct_simple_unaligned_layout(dtype)) {
-        return arraydescr_struct_list_str(dtype);
+        sub = arraydescr_struct_list_str(dtype);
     }
     else {
-        return arraydescr_struct_dict_str(dtype, includealignflag);
+        sub = arraydescr_struct_dict_str(dtype, includealignflag);
+    }
+    
+    /* If the data type has a nonstandard (subclassed) type, show it */
+    if(dtype->type_num == NPY_VOID && dtype->typeobj != &PyVoidArrType_Type) {
+        PyObject *typestr = PyUString_FromString(dtype->typeobj->tp_name);
+
+        PyObject *ret = PyUString_FromString("(");
```

Comment:
|They all return NULL or -1 if an exception occurs.

I wonder if `ret` should be checked for `NULL`? It's possible an error could occur and we wouldn't know it.